src/components/debounced-input.js

```diff
@@ -1,18 +1,23 @@
 import { Input } from 'antd';
 import PropTypes from 'prop-types';
-import React, { useEffect, useState } from 'react';
 import { useDebouncedCallback } from '../ui/hooks.js';
+import React, { useEffect, useRef, useState } from 'react';
 
 function DebouncedInput({ apiRef, timeLimit, elementType: Component, value, onChange, onBlur, onSearch, onPressEnter, ...props }) {
+  const actualValueRef = useRef(null);
   const [actualValue, setActualValue] = useState(value);
   const debouncedOnChange = useDebouncedCallback(onChange, timeLimit);
 
+  actualValueRef.current = actualValue;
+
   apiRef.current = {
     flush: () => debouncedOnChange.flush()
   };
 
   useEffect(() => {
-    setActualValue(value);
+    if (value !== actualValueRef.current) {
```

Comment:
We cannot (or maybe rather should not?) use `actualValue` in here, because that would make it a dependency of the `useEffect` and then the`useEffect` would be triggered on each key stroke (as `actualValue` changes). But this `useEffect` is here only as a guard if the prop changes the value programmatically (as opposed to user keyboard input).

Reply:
In other words, "d'oh!" 😅 Thanks for taking the time to correct my missunderstanding.